error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/JUnitToAssertJRules.java

```diff
@@ -0,0 +1,491 @@
+    void after(boolean actual, String message) {
+      assertThat(actual).withFailMessage(message).isFalse();
+    }
+  }
+
+  static final class AssertThatWithFailMessageSupplierIsFalse {
+    @BeforeTemplate
+    void before(boolean actual, Supplier<String> supplier) {
+      assertFalse(actual, supplier);
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(boolean actual, Supplier<String> supplier) {
+      assertThat(actual).withFailMessage(supplier).isFalse();
+    }
+  }
+
+  static final class AssertThatIsNull {
+    @BeforeTemplate
+    void before(Object actual) {
+      assertNull(actual);
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(Object actual) {
+      assertThat(actual).isNull();
+    }
+  }
+
+  static final class AssertThatWithFailMessageStringIsNull {
+    @BeforeTemplate
+    void before(Object actual, String message) {
+      assertNull(actual, message);
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(Object actual, String message) {
+      assertThat(actual).withFailMessage(message).isNull();
+    }
+  }
+
+  static final class AssertThatWithFailMessageSupplierIsNull {
+    @BeforeTemplate
+    void before(Object actual, Supplier<String> supplier) {
+      assertNull(actual, supplier);
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(Object actual, Supplier<String> supplier) {
+      assertThat(actual).withFailMessage(supplier).isNull();
+    }
+  }
+
+  static final class AssertThatIsNotNull {
+    @BeforeTemplate
+    void before(Object actual) {
+      assertNotNull(actual);
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(Object actual) {
+      assertThat(actual).isNotNull();
+    }
+  }
+
+  static final class AssertThatWithFailMessageStringIsNotNull {
+    @BeforeTemplate
+    void before(Object actual, String message) {
+      assertNotNull(actual, message);
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(Object actual, String message) {
+      assertThat(actual).withFailMessage(message).isNotNull();
+    }
+  }
+
+  static final class AssertThatWithFailMessageSupplierIsNotNull {
+    @BeforeTemplate
+    void before(Object actual, Supplier<String> supplier) {
+      assertNotNull(actual, supplier);
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(Object actual, Supplier<String> supplier) {
+      assertThat(actual).withFailMessage(supplier).isNotNull();
+    }
+  }
+
+  // XXX: Rewrite `org.junit.jupiter.api.Assertions.assertEquals`.
+  // XXX: Rewrite `org.junit.jupiter.api.Assertions.assertArrayEquals`.
+  // XXX: Rewrite `org.junit.jupiter.api.Assertions.assertIterableEquals`.
+  // XXX: Rewrite `org.junit.jupiter.api.Assertions.assertLinesMatch`.
+  // XXX: Rewrite `org.junit.jupiter.api.Assertions.assertNotEquals`.
```

Comment:
This listing isn't exhaustive (especially when we consider the various overloads), and making it so doesn't seem worth the hassle, if done manually. Let's call out the incompleteness at the start of the class, and defer further work to another PR.

Reply:
I started work on this idea [here](https://github.com/PicnicSupermarket/error-prone-support/compare/sschroevers/introduce-refaster-type-migration-annotation) (heavy WIP, and some changes should be moved to separate PRs, but the gist is there).

Reply:
Wow, that branch is awesome! Looks like some nice changes @Stephan202 !